### DISCUSSION TITLE
Improve user-friendliness

### DISCUSSION THREAD

**Discussion by MarvinLvn** [LAAC-LSCP/ChildProject#314] [Ideas]
/!\ I'll edit this post with new ideas when they'll come /!\

Hi there,

I'm gonna use the ChildProject to get some data. 
I'll take this opportunity to share a few thoughts about what may be improved instruction-wise.

First, here's what I'm gonna do : 

1) Get LENA Lyon recordings & metadata
2) Run the vtc
3) Extract adult speech

Relevant ressources Alex gave me (and the ones I found myself) : 

- The repo in itself : https://gin.g-node.org/EL1000/lyon
- The most-complete-but-quite-complicated doc : https://childproject.readthedocs.io
- The more-appealing-less-complicated doc : https://gin.g-node.org/EL1000/EL1000 

Ideas : 
1) When possible, git repos should be as self-sufficient as possible, I think they should minimally contain a `README.md` with instructions to a) get access to the repo (if needed) b) clone the repo c) get the metadata d) get the recordings. For instance, no instructions can be found in : https://gin.g-node.org/EL1000/lyon
Lazy people won't be willing to dig into the readthedocs as it is quite dense. I think it may prevent outsiders to use the pipeline

**Comment by alecristia** [comment]
thank you, so useful!

For the README we can adapt from this one: https://gin.g-node.org/EL1000/el1000

**Comment by MarvinLvn** [comment]
Definitely, this is by far the piece of documentation that I found the least repellent

**Comment by lucasgautheron** [comment]
Hello Marvin!

You're absolutely right.
READMEs for the datasets are still a WIP; the idea was to standardize them a little. suggestions welcome for the template here: https://gin.g-node.org/EL1000/EL1000/issues/11#issuecomment-1584

Note that the readthedocs version has conceptually different goal, which is to comprehensively document the package;
everything in EL1k should be specific to this dataset, whether it concerns data users (the majority of people, hopefully!) or maintainers (us).

**Reply by MarvinLvn** [reply]
Yep this what I understood, makes sense indeed !

**Comment by MarvinLvn** [comment]
Just launched the command `child-project process --threads 8 . standard basic --split=20:00:00 --format=wav --sampling=16000 --codec=pcm_s16le` to convert the recordings and I've noticed a couple of weird things.

1) First, not sure why this is useful in my case as I don't use ELAN/Praat + it changes the naming convention of the files, breaking some scripts I wrote to process the LENA lyon files.
2) If I list files under `recordings/raw`, I get something like :

```
10910_1               e20120724_105032_008109_1  e20120918_153305_008344_1  e20130314_125307_008344_1  e20130329_170857_008343_1  e20130817_183307_008066_1
COF_030504_1  LEN_010927_1               e20120726_112748_008015_1  e20120918_153528_008343_1  e20130318_152148_008066_1  e20130403_173734_008066_1  e20130825_190818_008344_1
```

These files don't have any extensions, but they appear to be .wav files (there are 63 files)

3) In `recordings/originals`, I have 54 .wav files. I don't fully understand the difference between `raw` and `originals`  (some of the .wav files are present in both folders, some others are not)

4) Now, let's list files under `recordings/converted/standard`, we have something like : 
```
COF_030504_1.000.wav
COF_030509_1.000.wav
COF_030509_2.000.wav
COF_030509_3.000.wav
```

I don't think the suffix is meant to be a float number. 
Also, why `COF_030509.wav` has been splitted in three files `COF_030509_1.000.wav`,  `COF_030509_2.000.wav`, `COF_030509_3.000.wav` of durations 5h57mn, 4h21mn, 1h43mn respectively ? I thought that I specified 20h in the `child-project process` command I ran

From what I understood child-project process command is supposed to split files if they're too long + convert them to 16khz, mono-channel, etc. 
In my case I wanted to ask to split files if they're longer than 20h, I don't think there are files longer than 20h. And, they're already sampled at 16khz, mono-channel etc.
 So I think the command doesn't have to run anything, it'd be nice to check if something needs to be done first, and then stop the program if no conversion is needed.

**Reply by alecristia** [reply]
regarding the names, I'm sorry, I thought I had cleaned that branch but clearly I had not. COF etc are the files from homebank which are wrong, and should not be used. They should be removed altogether. (Unless you're working in the main/master branch?)

**Reply by alecristia** [reply]
to clarify:
- I think some of the files you are finding may be there by mistake. In the commit history for the `newrecs` branch, mine was the latest commit, so I'll just go ahead and fix this oversight of mine. 
- another difficulty you have relates to another error of mine: there is no README explaining the contents of the repo. I'll do it now.
- my hunch is that you ARE looking at the main/master branch, so I'll merge the `newrecs` branch there, to avoid confusion in the future.

**Reply by alecristia** [reply]
no, confirmed -- you were looking at the main/master branch.
OK, I'll add a readme & merge mine with that, to avoid confusion in the future!

**Comment by lucasgautheron** [comment]
Hello!

I think the confusion comes from the fact you're trying to split recordings according to pauses in the recordings, and not every N hours, right ? (I don't know what you are trying to do, though -- but it was my understanding after talking with Alex that this is what you were doing?)
In any cayse, don't use `child-project process`'s `--split` switch, because it does not do what you're trying to do !

Just to explain the situation with `recordings/originals` and `recordings/raw`:
 
- `recordings/originals` contains the audio as it was one homebank. the naming convention was different than that of the version of the dataset I was previously given (not to mention attribution errors one homebank); also, they were not split according to pauses in the original recordings
- I wrote a script which you can find here (https://gin.g-node.org/EL1000/lyon/src/main/scripts/split.py) to rename the recordings from homebank and split them according to pauses; this is how the raw files into `recordings/raw` were derived

Btw: I think it's not desirable for you to take over work on Lyon, which is _really_ messy, without discussing it together before!

**Reply by alecristia** [reply]
@MarvinLvn I think looking at the script will confuse you more than help you. Let me repeat: we should NOT NOT NOT be using the versions from homebank at all. We should not make reference to their naming convention. They should be deleted, and forgotten! I'll write another answer to your comment above with what you need to do.

@lucasgautheron I asked Marvin to take over Lyon, precisely because I feel it's better to have two people working on something, so that the documentation & record of what happens is open and clear. So Marvin's getting stuck is a good sign for me - it means I didn't do my previous job well enough!

**Comment by alecristia** [comment]
Conceptually, though, you ask a few things:
- files without extensions: weird, they shouldn't happen, but they have the homebank naming convention, so my guess is that this problem will disappear when i fix the stuff above
- recordings can be split for two reasons: 
      1. because they are over 15h in length (since ELAN gets stuck with files over 15h). We use the split argument for this. I'd like then for --split to be followed by 15, and not 20, then, because the glitch with ELAN happens for files that last 15h. 
      2. because there are multiple recordings in the same LENA-exported wav file. This happens when a recording is paused. In Lyon, this happens (almost) always because the team launched/stopped a trial recording but didn't purge them from the device. It's a good idea to separate this wav into recordings so that eg the annotations make sense (they correspond to reasonable start/stop times, otherwise you'd need a "pause" within the vtc annotations). To this end, Lucas was using the script he linked to #311 . It sounds like I should look into this myself to avoid confusion due to the fact that that script bears on lyon but when lyon was using the incorrect homebank versions.

So I'll put up an update about this soon!

**Reply by lucasgautheron** [reply]
Alright! In that case, I will restrict my interventions to clarify what the package does or does not do, since it's related to the documentation I wrote for the package and not the documentation of Lyon by itself. This way you can still make sure that Lyon's documentation is self-consistent!
